### PR TITLE
feat: search bar component

### PR DESCRIPTION
**Pull request by alexgarel**
Initial work on the search-bar component

we have a search box and a button. It can be tested locally on http://search.localhost:8000/static/off.html.

At the moment it just send the request, it does not display the results (next results components).

### CODE REVIEW

File: frontend/src/search-ctl.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    pageSize: Number = 10;
+
+    @state()
+    _pageCount?: Number;
+
+    @state()
+    _results?: {}[];
+
+    @state()
+    _count?: Number;
```
Comment (Valimp) [inline frontend/src/search-ctl.ts:67]: Is it usefull to comment states like you do for property ?

File: frontend/src/search-ctl.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        this.search();
+      }
+    }
+
+    /**
+     * Launching search
+     */
+    async search() {
+      const response = await fetch(this._searchUrl());
+      const data = await response.json();
```
Comment (Valimp) [inline frontend/src/search-ctl.ts:141]: I don't understand why this is in two lines

Comment (Valimp) [reply frontend/src/search-ctl.ts:141]: I didn't understand why the two "await" were necessary.
I just got it 👍🏼

File: frontend/src/search-ctl.ts (hunk 1 above)
Comment (alexgarel) [reply frontend/src/search-ctl.ts:67]: Good point, I'll add them on next PR.

**Comment by alexgarel** [comment]
Merging, as it's work in progress, not displayed yet.

**Pull request merged**